src/main/java/wooteco/subway/service/LineService.java

```diff
@@ -29,41 +42,85 @@ public LineResponse createLine(@RequestBody LineRequest lineRequest) {
         }
-        Line newLine = lineDao.save(new Line(lineRequest.getName(), lineRequest.getColor()));
-        return new LineResponse(newLine.getId(), newLine.getName(), newLine.getColor());
+        Line createdLine = lineDao.save(new Line(lineRequest.getName(), lineRequest.getColor()));
+        sectionDao.save(new Section(createdLine.getId(), lineRequest.getUpStationId(), lineRequest.getDownStationId(), lineRequest.getDistance()));
+        List<StationResponse> stationResponses = generateStationResponsesByLineRequest(lineRequest);
+        return new LineResponse(createdLine.getId(), createdLine.getName(), createdLine.getColor(), stationResponses);
     }
 
     public List<LineResponse> showLines() {
         List<Line> lines = lineDao.findAll();
+        List<Station> stations = stationDao.findAll();
         return lines.stream()
-            .map(it -> new LineResponse(it.getId(), it.getName(), it.getColor()))
+            .map(line -> new LineResponse(line.getId(), line.getName(), line.getColor(), generateStationResponses(line, stations)))
             .collect(Collectors.toList());
     }
 
     public LineResponse showLine(Long id) {
         Line line = findLineById(id);
-        return new LineResponse(line.getId(), line.getName(), line.getColor());
+        List<Station> stations = stationDao.findAll();
+        List<StationResponse> stationResponses = generateStationResponses(line, stations);
+        return new LineResponse(line.getId(), line.getName(), line.getColor(), stationResponses);
+    }
+
+    public void updateLine(Long id, LineRequest lineRequest) {
+        Line originLine = findLineById(id);
+        validateUpdateDuplicateLine(originLine, lineRequest);
+        lineDao.update(originLine, new Line(id, lineRequest.getName(), lineRequest.getColor()));
+    }
+
+    public void deleteLine(Long id) {
+        Line line = findLineById(id);
+        sectionDao.deleteByLineId(id);
+        lineDao.delete(line);
+    }
+
+    public void addSection(long id, SectionRequest sectionRequest) {
+        List<Section> originSectionList = sectionDao.findByLineId(id).getSections();
+        Sections updateSections = new Sections(originSectionList);
+        Section addSection = new Section(id, sectionRequest.getUpStationId(), sectionRequest.getDownStationId(), sectionRequest.getDistance());
+        updateSections.add(addSection);
+        compareWithDao(originSectionList, updateSections.getSections());
+    }
+
+    public void deleteSection(long id, Long stationId) {
+        List<Section> originSectionList = sectionDao.findByLineId(id).getSections();
+        Sections updateSections = new Sections(originSectionList);
+        updateSections.remove(stationId);
+        compareWithDao(originSectionList, updateSections.getSections());
+    }
+
+    private List<StationResponse> generateStationResponsesByLineRequest(LineRequest lineRequest) {
```

Comment:
이미 인자를 통해 충분히 lineRequest를 받는다고 드러내요!

```suggestion
    private List<StationResponse> generateStationResponses(LineRequest lineRequest) {
```

Reply:
List<StationResponse>를 생성하는 함수에서 각각`LineRequest`, `List<Station>`로 인자가 달라 이름을 분리해봤었습니다.
생각해보니 동일한 메서드명으로 인자가 다른 여러 멧서드를 만들 수 있다는 사실을 잠시 까먹고있었네요;;;
`메서드명과 인자의 의미 중복`에 대해서 생각해보면서 메서드명을 작성하도록하겠습니다 ㅎㅎ

Reply:
네 요 부분이 메서드 명을 센스있게 작성할 수 있는 팁이기에 머리에 남겨두시면 좋습니다!